_specifications/lsp/3.18/types/diagnostic.md

```diff
@@ -34,10 +34,22 @@ export interface Diagnostic {
 	source?: string;
 
 	/**
-	 * The diagnostic's message.
+	 * The diagnostic's message, e.g. 'Unknown type name "int_8_t"'.
 	 */
 	message: string;
 
+	/**
+	 * An optional property to describe this diagnostic's message with more
+	 * details. For example:
+	 * The type name "int_8_t" is unknown. Did you mean any of these types?
+	 * - int
+	 * - int8_t
+	 * Read more about the built-in types [here](file:///path/to/docs.md).
+	 *
+	 * @since 3.18.0
+	 */
+	longMessage?: string | MarkupContent;
```

Comment:
The naming of the property can suggest that it's an alternative version of `message`. Not sure if the intention would be that clients show both or choose one.

If the intention is to show more then maybe this should be called `description`...

Reply:
Calling it `description` would be too similar to the property called `codeDescription`, which is a general/static description of the diagnostic.

In contrast, I thought about calling it `longMessage` because it is similar to `message` but provides more detail. Perhaps something like `messageDetails` would be more appropriate. I think that would infer that `message` is still important.